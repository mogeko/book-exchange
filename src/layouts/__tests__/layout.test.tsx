import { render, screen } from "@testing-library/react";
import "@testing-library/jest-dom";
import Layout, { DefaultLayout } from "@/layouts/layout";

describe("Layout", () => {
  it("should render correctly", () => {
    render(
      <Layout>
        <div>hello, world!</div>
      </Layout>
    );
    expect(screen.getByText(/hello, world!/)).toBeInTheDocument();
  });

  it("snapshot a Layput container", () => {
    const { container } = render(
      <Layout>
        <div>hello, world!</div>
      </Layout>
    );
    expect(container).toMatchSnapshot();
  });

  it("should render correctly with Header and Footer", () => {
    render(
      <DefaultLayout>
        <div>hello, world!</div>
      </DefaultLayout>
    );
    expect(screen.getByText(/hello, world!/)).toBeInTheDocument();
    expect(
      screen.getByText("Copyright Â© 2022, All right reserved by Zheng Junyi.")
    ).toBeInTheDocument();
    expect(
      screen.getByRole("heading", {
        name: /Bookworm/i,
      })
    ).toBeInTheDocument();
  });
});
