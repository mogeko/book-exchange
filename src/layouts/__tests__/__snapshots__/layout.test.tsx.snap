// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Layout should render correctly 1`] = `
<div>
  <div
    class="flex flex-col min-h-screen min-w-full w-full"
  >
    <div>
      hello, world!
    </div>
  </div>
</div>
`;

exports[`Layout should render correctly with Header and Footer 1`] = `
<div>
  <div
    class="drawer"
  >
    <input
      aria-hidden="true"
      class="drawer-toggle"
      id="mobile-menu"
      type="checkbox"
    />
    <div
      class="drawer-content"
    >
      <div
        class="flex flex-col min-h-screen min-w-full w-full"
      >
        <div
          class="min-w-full"
        >
          <div
            class="hidden h-7 bg-black justify-between sm:flex"
            data-theme="dracula"
          >
            <nav
              class="navbar min-h-0 flex-"
            >
              <a
                class="btn btn-link btn-xs gap-1 text-gray-400 hover:text-primary focus:text-primary-focus"
                href="/#"
              >
                My library
              </a>
              <a
                class="btn btn-link btn-xs gap-1 text-gray-400 hover:text-primary focus:text-primary-focus"
                href="/#"
              >
                Ranking
              </a>
              <a
                class="btn btn-link btn-xs gap-1 text-gray-400 hover:text-primary focus:text-primary-focus"
                href="/#"
              >
                Quote
              </a>
              <a
                class="btn btn-link btn-xs text-gray-400 hover:text-primary focus:text-primary-focus"
                href="/random"
              >
                I'm Feeling Lucky!
              </a>
            </nav>
            <div
              class="justify-end"
            >
              <div
                class="normal-case hidden sm:block"
              >
                <a
                  class="btn btn-link btn-xs w-40 flex-nowrap"
                  href="/login"
                >
                  Sign in / Sign up
                </a>
              </div>
              <a
                class="btn btn-ghost btn-circle avatar sm:hidden"
                href="/login"
              >
                <svg
                  aria-hidden="true"
                  class="h-6 w-6"
                  fill="currentColor"
                  height="1em"
                  stroke="currentColor"
                  stroke-width="0"
                  viewBox="0 0 20 20"
                  width="1em"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <path
                    clip-rule="evenodd"
                    d="M10 9a3 3 0 100-6 3 3 0 000 6zm-7 9a7 7 0 1114 0H3z"
                    fill-rule="evenodd"
                  />
                </svg>
              </a>
            </div>
          </div>
          <div
            class="bg-inherit p-2 md:p-0 md:bg-base-300"
          >
            <header
              class="navbar p-0 m-auto rounded-md max-w-screen-lg bg-base-300 md:rounded-none"
            >
              <div
                class="navbar-start sm:hidden"
              >
                <label
                  aria-label="Drawer Button"
                  class="btn btn-ghost btn-circle"
                  for="mobile-menu"
                  role="button"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M3 5a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 10a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1zM3 15a1 1 0 011-1h12a1 1 0 110 2H4a1 1 0 01-1-1z"
                      fill-rule="evenodd"
                    />
                  </svg>
                </label>
              </div>
              <div
                class="navbar-center sm:navbar-start"
              >
                <div
                  class="flex justify-start"
                >
                  <a
                    class="btn gap-1 btn-ghost normal-case"
                    href="/"
                  >
                    <div
                      class="w-6 relative aspect-square"
                    >
                      <span
                        style="box-sizing: border-box; display: block; overflow: hidden; background: none; opacity: 1; border: 0px; margin: 0px; padding: 0px; position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px;"
                      >
                        <img
                          alt="logo"
                          data-nimg="fill"
                          decoding="async"
                          src="data:image/gif;base64,R0lGODlhAQABAIAAAAAAAP///yH5BAEAAAAALAAAAAABAAEAAAIBRAA7"
                          style="position: absolute; top: 0px; left: 0px; bottom: 0px; right: 0px; box-sizing: border-box; padding: 0px; margin: auto; display: block; width: 0px; height: 0px; min-width: 100%; max-width: 100%; min-height: 100%; max-height: 100%;"
                        />
                        <noscript />
                      </span>
                    </div>
                    <h1
                      class="text-xl font-bold text-base-content"
                    >
                      Bookworm
                    </h1>
                    <span
                      class="sr-only"
                    >
                      Logo
                    </span>
                  </a>
                </div>
              </div>
              <div
                class="navbar-end"
              >
                <div
                  class="hidden md:block relative"
                >
                  <svg
                    class="h-6 w-6 absolute top-3 left-3 text-gray-400"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <div
                    class="form-control"
                  >
                    <input
                      aria-label="Search Bar"
                      class="input bg-base-300 text-xl w-96 pl-10"
                      placeholder="Search..."
                      role="search"
                      type="text"
                    />
                  </div>
                </div>
                <a
                  class="btn btn-ghost btn-circle md:hidden"
                  href="/#"
                  role="button"
                >
                  <svg
                    aria-hidden="true"
                    class="h-6 w-6"
                    fill="currentColor"
                    height="1em"
                    stroke="currentColor"
                    stroke-width="0"
                    viewBox="0 0 20 20"
                    width="1em"
                    xmlns="http://www.w3.org/2000/svg"
                  >
                    <path
                      clip-rule="evenodd"
                      d="M8 4a4 4 0 100 8 4 4 0 000-8zM2 8a6 6 0 1110.89 3.476l4.817 4.817a1 1 0 01-1.414 1.414l-4.816-4.816A6 6 0 012 8z"
                      fill-rule="evenodd"
                    />
                  </svg>
                  <span
                    class="sr-only"
                  >
                    Search
                  </span>
                </a>
                <div
                  class="sm:hidden"
                >
                  <div
                    class="normal-case hidden sm:block"
                  >
                    <a
                      class="btn btn-link btn-xs w-40 flex-nowrap"
                      href="/login"
                    >
                      Sign in / Sign up
                    </a>
                  </div>
                  <a
                    class="btn btn-ghost btn-circle avatar sm:hidden"
                    href="/login"
                  >
                    <svg
                      aria-hidden="true"
                      class="h-6 w-6"
                      fill="currentColor"
                      height="1em"
                      stroke="currentColor"
                      stroke-width="0"
                      viewBox="0 0 20 20"
                      width="1em"
                      xmlns="http://www.w3.org/2000/svg"
                    >
                      <path
                        clip-rule="evenodd"
                        d="M10 9a3 3 0 100-6 3 3 0 000 6zm-7 9a7 7 0 1114 0H3z"
                        fill-rule="evenodd"
                      />
                    </svg>
                  </a>
                </div>
              </div>
            </header>
          </div>
        </div>
        <main
          class="flex-1 max-w-screen-lg w-full mx-auto p-6 md:p-4"
        >
          <div>
            hello, world!
          </div>
        </main>
        <footer
          aria-label="Footer"
          class="footer footer-center p-4 bg-base-300 text-base-content"
        >
          <div>
            <p
              aria-label="copyright"
              role="presentation"
            >
              Copyright Â© 
              2022
              , All right reserved
               by Zheng Junyi
              .
            </p>
          </div>
        </footer>
      </div>
    </div>
    <div
      class="drawer-side"
    >
      <label
        class="drawer-overlay"
        for="mobile-menu"
      />
      <ul
        aria-label="Drawer Menu"
        class="menu p-4 overflow-y-auto w-80 bg-base-200 text-base-content"
        role="menu"
      >
        <li
          role="menuitem"
        >
          <a
            class="justify-between"
            href="/#"
          >
            <span>
              My library
            </span>
          </a>
        </li>
        <li
          role="menuitem"
        >
          <a
            class="justify-between"
            href="/#"
          >
            <span>
              Ranking
            </span>
          </a>
        </li>
        <li
          role="menuitem"
        >
          <a
            class="justify-between"
            href="/#"
          >
            <span>
              Quote
            </span>
          </a>
        </li>
      </ul>
    </div>
  </div>
</div>
`;
